src/Configuration.php

```diff
@@ -375,6 +378,11 @@ public function addCustomHydrationMode(string $modeName, string $hydrator): void
      */
     public function setClassMetadataFactoryName(string $cmfName): void
     {
+        $reflectionClass = new ReflectionClass($cmfName);
+        if (! $reflectionClass->implementsInterface(ClassMetadataFactoryInterface::class)) {
+            throw InvalidClassMetadataFactory::create($cmfName);
+        }
```

Comment:
This would introduce an exception when there was not one before.

Reply:
Prior to this PR a call with a class different from `Doctrine\ORM\Mapping\ClassMetadataFactory` would cause a TypeError. So yes, it would not throw an exception here, but in a different (and completely unrelated) point of the code, which can be harder to debug.